app/cdap/components/Replicator/utilities/index.ts

```diff
@@ -383,3 +384,23 @@ export function getTableDisplayName(row) {
 
   return displayName;
 }
+
+/**
+ * useDebounce returns a debounced function
+ * @example
+ *  const debounce = useDebounce(100);
+ *
+ *  debounce(() => {
+ *   functionToExecute();
+ *  });
+ * @param ms time in ms
+ * @returns function that takes a function as its argument and executes after the debouncea
+ * eg
+ */
+export const useDebounce = (ms: number) =>
```

Comment:
I think this is not needed anymore?

Reply:
oh weird i thought i removed it - my bad

Reply:
to clarify - with an empty dependency array and mimicking componentWillUnmount it doesn't have access to the newest state